content/geometry/CircleLine.h

```diff
@@ -0,0 +1,26 @@
+/**
+ * Author: Victor Lecomte, chilli
+ * Date: 2019-10-29
+ * License: CC0
+ * Source: https://vlecomte.github.io/cp-geo.pdf
+ * Description: Finds the intersection between a circle and a line. Returns a
+ * vector of either 0, 1, or 2 intersection points. P is intended to be
+ * Point<double>
+ * Status:
+ */
+
+#pragma once
+
+#include "Point.h"
+#include "lineDistance.h"
+#include "LineProjectionReflection.h"
+
+template<class P>
+vector<P> circleLine(P c, double r, P a, P b) {
+	double h2 = r * r - lineDist(a, b, c) * lineDist(a, b, c);
```

Comment:
`lineDist(a, b, c) * lineDist(a, b, c)` is kinda terrible; for one it means the `h2 == 0` comparison isn't meaningful since `h2` has floating point imprecision from the lineDist sqrt. Should be easy to rewrite without the sqrt, right?